homeassistant/components/ecowitt/entity.py

```diff
@@ -0,0 +1,48 @@
+"""The Ecowitt Weather Station Entity."""
+from __future__ import annotations
+
+import time
+
+from aioecowitt import EcoWittSensor
+
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import DOMAIN
+
+
+class EcowittEntity(Entity):
+    """Base class for Ecowitt Weather Station."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, sensor: EcoWittSensor) -> None:
+        """Construct the entity."""
+        self.ecowitt: EcoWittSensor = sensor
+
+        self._attr_name = sensor.name
```

Comment:
Not needed as this is provided by the entity description (or it can be removed from entity description).